velox/type/DecimalUtil.h

```diff
@@ -278,7 +278,9 @@ class DecimalUtil {
       uint8_t /*bRescale*/) {
     VELOX_USER_CHECK_NE(b, 0, "Division by zero");
     int resultSign = 1;
-    R unsignedDividendRescaled(a);
+    using IntermediateType =
```

Comment:
Would you add a comment for this method to explain what it does and what are the template parameters?

Does it compute a / b, then rounds (if noRoundUp == false)?

Why bRecale is not used? Can it be removed? What is aRescale parameter?

What are valid types for A, B, R? Do these have to be either int64_t or int128_t? Let's document that and add a static_assert.

Would you add a test to velox/functions/sparksql/tests/DecimalUtilTest.cpp directly?